tests/integration/construct_component_methods/testcomponent-python/testcomponent.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2016-2021, Pulumi Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional
+import sys
+
+import pulumi
+import pulumi.provider as provider
+
+
+class Component(pulumi.ComponentResource):
+    def __init__(self,
+                 resource_name: str,
+                 opts: Optional[pulumi.ResourceOptions] = None,
+                 first: Optional[pulumi.Input[str]] = None,
+                 second: Optional[pulumi.Input[str]] = None) -> None:
+
+        args = {
+            "first": first,
+            "second": second,
+        }
+        super().__init__("testcomponent:index:Component", resource_name, args, opts)
+
+        if opts and opts.urn:
+            return
+
+        self.register_outputs(args)
+
+    @property
+    @pulumi.getter
+    def first(self) -> pulumi.Output[str]:
+        return pulumi.get(self, "first")
+
+    @property
+    @pulumi.getter
+    def second(self) -> pulumi.Output[str]:
+        return pulumi.get(self, "second")
+
+    def get_message(self, name: pulumi.Input[str]) -> pulumi.Output[str]:
```

Comment:
I'm hoping/guessing the `self` here exercises the "do not wrap in output" Resource  reference path I had qns on earlier, if so we're good with testing it..

Reply:
Right. It's really for line 87-89:

```python
        comp: Component = args["__self__"]
         outputs = {
             "message": comp.get_message(args["name"])
         }
```

So you can call `comp.get_message(...)` and don't have to do `comp.apply(lambda v: v.get_message(...))`.